src/context_free.ml

```diff
@@ -484,14 +503,16 @@ class map_top_down ?(expect_mismatch_handler = Expect_mismatch_handler.nop)
         | Pexp_extension _ ->
             map_node EC.expression expression
               (fun _ e -> return e)
-              e.pexp_loc base_ctxt e
+              e.pexp_loc base_ctxt e ~embed_errors
         | _ -> return e
       in
       expanded >>= fun e ->
       let expand_constant kind char text =
         match Hashtbl.find_opt constants (char, kind) with
         | None -> super#expression base_ctxt e
-        | Some expand -> self#expression base_ctxt (expand e.pexp_loc text)
+        | Some expand -> (
+            try self#expression base_ctxt (expand e.pexp_loc text)
```

Comment:
Only the call to `expand`, and not the call to `self#expression`, should be inside the `try ... with`.
Otherwise, we would risk catching exceptions that were not meant to be caught here. Also, I think it is more readable to `try ... with` only the expression that can raise.